streaming/dataset.py

```diff
@@ -174,98 +174,112 @@ class StreamingDataset(Array, IterableDataset):
-            ``remote``/``local``. Defaults to ``None``.
-        local (str, optional): Local working directory to download shards to. This is where shards
-            are cached while they are being used. Uses a temp directory if not set.
-            StreamingDataset uses either ``streams`` or ``remote``/``local``. Defaults to ``None``.
-        split (str, optional): Which dataset split to use, if any. If provided, we stream from/to
-            the ``split`` subdirs of  ``remote`` and ``local``. Defaults to ``None``.
-        download_retry (int): Number of download re-attempts before raising an error. Defaults to
-            ``2``.
-        download_timeout (str | float): Time in seconds to wait for a file download to complete
-            before raising an error. Streaming duration shorthand (e.g., ``1m23s``) is also
-            accepted. Defaults to ``1m``.
-        hash_algos (str | Sequence[str], optional): Ranked list of hashing algorithms to try.
-            Defaults to ``None``.
-        validate_hash (str, optional): Deprecated. See ``hash_algos``. Defaults to ``None``.
-        keep_old_phases (str, optional): Which old phases of shard files to cache (until shard
-            eviction). If set, must be one of ``nil``, ``src``, or ``all``. Defaults to ``None``,
-            which uses ``keep_zip``, falling back to ``nil``.
-        keep_zip (bool, optional): Deprecated. See ``keep_old_phases``. Defaults to ``None``.
         epoch_size (Union[str, int], optional): Number of samples to draw per epoch balanced
             across all streams. If ``None``, takes its value from the total number of underlying
             samples. Provide this field if you are weighting streams relatively to target a larger
             or smaller epoch size. Defaults to ``None``. Can also take in human-readable number
             abbreviations (e.g., ``"100k"``, ``"64M"``, ``"77b"``, etc). Defaults to ``None``.
+        streams (Sequence[Stream], optional): One or more streams to stream/cache samples from,
+            which may be upsampled or downsampled. StreamingDataset uses either ``streams`` or
+            ``remote``/``local``. Defaults to ``None``.
+        remote (str, optional): Remote path to stream the dataset from. If ``None``, dataset must
+            be complete locally. Defaults to ``None``.
+        local (str, optional): Local working directory to stream the dataset to. Uses a
+            deterministically-calculated temp directory if not set. Defaults to ``None``.
+        split (str, optional): Which dataset split sub-path to use, if any. Defaults to ``None``.
+        index_size (str | int, optional): Expected index size in bytes. Defaults to ``None``.
+        index_hashes (Dict[str, str], optional): Available index hashes. This is a mapping of hash
+            algo name to expected hex digest. Defaults to ``None``.
+        allow_schema_mismatch (bool): If ``True``, continue if schemas mismatch across
```

Comment:
Can you explain a bit in the doc what the schema actually is?

Reply:
How about this?

I figure `schema` is the more general/more proper terminology over `columns`, but in our case it's just referring to our columns...